src/api/middleware/company.js

```diff
@@ -85,6 +85,45 @@ export const profileComplete = async (req, res, next) => {
     return next();
 };
 
+export const canAccessProfile = (companyId) => async (req, res, next) => {
+    const company = await new CompanyService().findById(companyId, true);
+
+    const notFound = () =>
+        new APIError(
+            HTTPStatus.UNPROCESSABLE_ENTITY,
+            ErrorTypes.VALIDATION_ERROR,
+            [
+                {
+                    value: companyId,
+                    msg: ValidationReasons.COMPANY_NOT_FOUND(companyId),
+                    param: "companyId",
+                    location: "params",
+                },
+            ]
```

Comment:
@ttoino can you explain why you are setting the error directly here? I recall you telling me that already but I've forgotten it.

Reply:
I wanted the endpoint to return the same error as when the company does not exist, right now the only way is to do that (I also hate it), but after #282 we could just return a 404.

Reply:
I find this a bit confusing but it could be improved in #282